mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -306,12 +307,18 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    fit(dataset, handlePersistence)
+  }
+  protected def fit(dataset: Dataset[_], handlePersistence: Boolean): KMeansModel = {
```

Comment:
Still wrong -- no space, and now no `@Since` at all. This is `@Since("2.2.0")` if it goes in.